noria/src/data.rs

```diff
@@ -387,17 +387,7 @@ impl<'a> From<&'a Literal> for DataType {
 
 impl From<Literal> for DataType {
     fn from(l: Literal) -> Self {
-        match l {
-            Literal::Null => DataType::None,
-            Literal::Integer(i) => (i as i64).into(),
-            Literal::String(s) => s.as_str().into(),
```

Comment:
It's pretty sad that we can't re-use the storage here, but it is what it is :shrug: